lib/maths/unittest/CBoostedTreeTest.cc

```diff
@@ -1306,7 +1306,7 @@ BOOST_AUTO_TEST_CASE(testEstimateMemoryUsedByTrain) {
         BOOST_TEST_REQUIRE(instrumentation.maxMemoryUsage() < estimatedMemory);
     }
 }
-
+// TODO: test for handle fatal (memory-limit-circuit-breaker)
```

Comment:
I think it is worth adding this test as part of this change. You just need to override the fatal handler so it doesn't call `std::terminate`, i.e.
```
TStrVec errors;
auto errorHandler = [&errors](std::string error) { errors.push_back(error); };
core::CLogger::CScopeSetFatalErrorHandler scope{errorHandler};
```
See for example `CDataFrameAnalysisSpecificationTest/testCreate`

Reply:
Thank you for the tip. I created tests in `CDataFrameAnalyzerTrainingTest` and `CDataFrameAnalyzerOutlierTest` to check for errors.